### ISSUE TITLE
Check for pylint pragmas in `__init__.py`

### ISSUE THREAD

**Issue by chaosflaws** [pylint-dev/pylint#9496]
### Current problem

I would like to disable different pylint rules in my application code and my test code (specifically messages about certain missing docstrings).

### Desired solution

There is an open issue for that (#2712), but I suppose this won't be supported soon. An easier solution to the problem I am facing would be to parse `__init__.py` files for pylint pragmas and, if present, apply them to all modules in that package.

### Additional context

IMHO, this should greatly reduce the need for #2712, since with this addition, all file- and block-related exclusions, file-based exclusions and directory-based exclusions would be supported.

**Comment by Pierre-Sassoulas** [comment]
We have a merge request open for per dir pylint configuration files which imo are better because it would be harder to miss that a package is affected by it and it would permit to configure more than just disable or enable. It would be easy to miss an `__init__.py`, assume it's empty and be surprised by it's effect.

**Comment by chaosflaws** [comment]
Agreed, #9395 is the better solution, and is already in the works :-)
